src/main/java/uk/gov/ida/verifyserviceprovider/services/ResponseService.java

```diff
@@ -72,16 +70,30 @@ public TranslatedResponseBody convertTranslatedResponseBody(
                 return responderCodeTranslator.translateResponderCode(statusCode);
             case StatusCode.SUCCESS:
                 List<Assertion> assertions = assertionDecrypter.decryptAssertions(validatedResponse);
+                if (assertionsContainEidasUnsignedAssertionsResponse(assertions)) {
+                    UnsignedAssertionsResponseHandler handler = unsignedAssertionsResponseHandler.get();
+
+                    ValidatedResponse validatedCountryResponse = handler.getValidatedResponse(assertions.get(0), expectedInResponseTo);
+                    assertions = handler.decryptAssertion(validatedCountryResponse, assertions.get(0));
+                }
                 return assertionTranslator.translateSuccessResponse(assertions, expectedInResponseTo, expectedLevelOfAssurance, entityId);
             default:
                 throw new SamlResponseValidationException(String.format("Unknown SAML status: %s", statusCode.getValue()));
         }
     }
 
-    private ValidatedResponse getValidatedResponse (Response response) {
-        String issuer = response.getIssuer().getValue();
-        return eidasValidatorFactory.isPresent() && !issuer.equals(hubEntityId) ?
-                eidasValidatorFactory.get().getValidatedResponse(response) :
-                responseSignatureValidator.validate(response, SPSSODescriptor.DEFAULT_ELEMENT_NAME);
+    private boolean assertionsContainEidasUnsignedAssertionsResponse(List<Assertion> assertions) {
+        if (assertions == null || assertions.size() != 1) { return false; }
```

Comment:
In fact, perhaps this method could state its assumptions. (If we were to comment on them.)

Reply:
Did the other thing, so hopefully it's clearer now.